### PR TITLE
Add edc40 keyboard

### PR DESCRIPTION
**Pull request by ojthetiny**
edc40, an affordable, 3d printed, low profile everyday carry board

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/edc40/rules.mk [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/edc40/rules.mk:11]:
```suggestion
```

File: keyboards/edc40/rules.mk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+
```
Comment (fauxpark) [inline keyboards/edc40/rules.mk:32]:
```suggestion
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
UNICODE_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/edc40/info.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    }
+}
```
Comment (fauxpark) [inline keyboards/edc40/info.json:10]:
```suggestion
{
    "keyboard_name": "edc40",
    "url": "",
    "maintainer": "OJ",
    "width": 12,
    "height": 4,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"x": 0, "y": 0},
                {"x": 1, "y": 0},
                {"x": 2, "y": 0},
                {"x": 3, "y": 0},
                {"x": 4, "y": 0},
                {"x": 5, "y": 0},
                {"x": 6, "y": 0},
                {"x": 7, "y": 0},
                {"x": 8, "y": 0},
                {"x": 9, "y": 0},
                {"x": 10, "y": 0},
                {"x": 11, "y": 0},

                {"x": 0, "y": 1},
                {"x": 1.25, "y": 1},
                {"x": 2.25, "y": 1},
                {"x": 3.25, "y": 1},
                {"x": 4.25, "y": 1},
                {"x": 5.25, "y": 1},
                {"x": 6.25, "y": 1},
                {"x": 7.25, "y": 1},
                {"x": 8.25, "y": 1},
                {"x": 9.25, "y": 1},
                {"x": 10.25, "y": 1},

                {"x": 0, "y": 2},
                {"x": 1.75, "y": 2},
                {"x": 2.75, "y": 2},
                {"x": 3.75, "y": 2},
                {"x": 4.75, "y": 2},
                {"x": 5.75, "y": 2},
                {"x": 6.75, "y": 2},
                {"x": 7.75, "y": 2},
                {"x": 8.75, "y": 2},
                {"x": 9.75, "y": 2},

                {"x": 0, "y": 3},
                {"x": 1.25, "y": 3},
                {"x": 4.75, "y": 3},
                {"x": 7.75, "y": 3},
                {"x": 9, "y": 3}
            ]
        }
    }
}
```
Just noticed the lint check wasn't happy about this.
Is this correct? (you can preview it in Configurator by pressing Ctrl+Shift+I)
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/edc40/info.json (hunk 3 above)
Comment (ojthetiny) [reply keyboards/edc40/info.json:10]: hmm, spacing seems off, I need to fix the info.json file, I assume that's what the configurator pulls from

**Review by fauxpark — APPROVED** [review]

File: keyboards/edc40/info.json (hunk 3 above)
Comment (ojthetiny) [reply keyboards/edc40/info.json]: now it's all good

File: keyboards/edc40/config.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x5454 // TT
```
Comment (noroadsleft) [inline keyboards/edc40/config.h:21]: Another contributor uses `0x5454` as their Vendor ID.

```suggestion
#define VENDOR_ID 0x4F4A // "OJ"
```

File: keyboards/edc40/info.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{
+    "keyboard_name": "edc40",
+    "url": "",
+    "maintainer": "OJ",
```
Comment (noroadsleft) [inline keyboards/edc40/info.json:4]: This is meant to reference your GitHub username.

```suggestion
    "maintainer": "ojthetiny",
```

File: keyboards/edc40/readme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# EDC40
+
+An everyday carry 40% keyboard
+
+* Keyboard Maintainer: OJ
```
Comment (noroadsleft) [inline keyboards/edc40/readme.md:5]: Quick link to your GitHub profile.

```suggestion
* Keyboard Maintainer: [OJ](https://github.com/ojthetiny)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Just a couple small tweaks to improve the user experience.

It'll also be great to get a link to the product listing page on the readme once this board goes on sale.

File: keyboards/edc40/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    L1,
+    L2
+};
+
+const uint16_t PROGMEM keymaps [][MATRIX_ROWS][MATRIX_COLS] = {
+    [L0] = LAYOUT( /* Base Layer */
+      KC_ESC,  KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC,
+      KC_CAPS,  KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L,    KC_ENT,
+      KC_LSFT,   KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM,    KC_DOT,
+      KC_CTRL,  KC_LGUI,             LT(L1, KC_SPC),              KC_RALT, MO(L2)
```
Comment (noroadsleft) [inline keyboards/edc40/keymaps/default/keymap.c:29]:
```suggestion
      KC_LCTL,  KC_LGUI,             LT(L1, KC_SPC),              KC_RALT, MO(L2)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**